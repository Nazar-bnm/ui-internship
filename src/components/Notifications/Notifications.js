import React, { Component } from 'react';
// import cx from 'classnames';
// import PropTypes from 'prop-types';

import './Notifications.scss';

class Notifications extends Component {
  // constructor() {
  //   super();


  // }

  render() {
    return (
      <div>Example text</div>
    );
  }
}


export default Notifications;
