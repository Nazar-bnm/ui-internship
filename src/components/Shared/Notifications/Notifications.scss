@import '../../../scss/global_styles';



.notifications {
  &__wrapper {
    position: fixed;
    top: 95px;
    right: 20px;
    z-index: 999;
  }

  display: block;
  color: $white;
  padding: 16px;
  margin: 5px;
  opacity: .8;
  transition: top 2s ease;
  width: 200px;
  text-align: center;
  min-height: 55px;
  box-sizing: border-box;
  border-radius: 5px;
  cursor: pointer;
  line-height: 1.2em;
  
  &__ntf-box {
    @include flex($justify: space-around, $wrap: nowrap, $align: center);
  }

  &__title {
    text-transform: uppercase;
  }

  &.change {
    background-color: #444;
  }

  &.info {
    background-color: #3498db;
  }

  &.success {
    background-color: #2ecc71;
  }

  &.warning {
    background-color: #e67e22;
  }

  &.error {
    background-color: #e74c3c;
  }

  &.custom {
    background-color: purple;
  }
}
