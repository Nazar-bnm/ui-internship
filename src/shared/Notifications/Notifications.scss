@import '../../scss/global_styles';



.notifications {
  &__wrapper {
    position: fixed;
    top: 95px;
    right: 20px;
    z-index: 105;
  }

  display: block;
  padding: 16px;
  opacity: .8;
  transition: top 2s ease;
  margin: 5px;
  border-radius: 5px;
  box-sizing: border-box;
  width: 200px;
  min-height: 55px;
  cursor: pointer;
  line-height: 1.2em;
  text-align: center;
  color: $white;
  
  &__notification-box {
    @include flex($justify: space-around, $wrap: nowrap, $align: center);
  }

  &__title {
    text-transform: uppercase;
  }

  &.change {
    background-color: $notification-change-grey;
  }

  &.info {
    background-color: $notification-info-blue;
  }

  &.success {
    background-color: $notification-success-green;
  }

  &.warning {
    background-color: $notification-warning-orange;
  }

  &.error {
    background-color: $notification-error-red;
  }

  &.custom {
    background-color: $notification-custom-purple;
  }
}
