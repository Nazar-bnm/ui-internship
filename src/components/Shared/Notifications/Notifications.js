import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import { NOTIFICATION_TYPES } from '../../../constants';
import './Notifications.scss';

export const CN = 'notifications';

class Notifications extends Component {
  constructor(props) {
    super(props);

    this.hideMessage = this.hideMessage.bind(this);
  }

  hideMessage() {
    const { isHideNotification } = this.props;
    isHideNotification();
  }

  render() {
    const {
      type,
      message,
      title,
      isShowNotification
    } = this.props;

    return isShowNotification && (
      <div
        className={cx(CN, type)}
        color={type}
        onClick={this.hideMessage}
      >
        <div onClick={this.onNotificationClick}>
          {title && <h4 className="title">{title}</h4>}
          <div className={`${CN}__notification-message-box`} role="alert">
            <div className={`${CN}__message`}>{message}</div>
          </div>
        </div>
      </div>
    );
  }
}

Notifications.propTypes = {
  type: PropTypes.oneOf([
    NOTIFICATION_TYPES.CHANGE,
    NOTIFICATION_TYPES.INFO,
    NOTIFICATION_TYPES.SUCCESS,
    NOTIFICATION_TYPES.WARNING,
    NOTIFICATION_TYPES.ERROR,
    NOTIFICATION_TYPES.CUSTOM
  ]),
  title: PropTypes.string,
  message: PropTypes.string
};

Notifications.defaultProps = {
  type: NOTIFICATION_TYPES.INFO,
  title: 'Info',
  message: 'Information is processing'
};

export default Notifications;
