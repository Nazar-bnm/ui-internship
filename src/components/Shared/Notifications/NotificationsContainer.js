import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Notifications from './Notifications';
import { isHideNotification, removeItemNotification } from '../../../actions/actionsNotifications';

const mapStateToProps = (state) => ({
  notifications: state.reducerNotifications
});

const mapDispatchToProps = (dispatch) => ({
  isHideNotification: bindActionCreators(isHideNotification, dispatch),
  removeItemNotification: bindActionCreators(removeItemNotification, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(Notifications);
