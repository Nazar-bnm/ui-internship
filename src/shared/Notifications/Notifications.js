import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import './Notifications.scss';

export const CN = 'notifications';

const notificationIcons = {
  info: {
    type: 'info',
    icon: 'info icon'
  },
  success: {
    type: 'success',
    icon: 'check circle '
  },
  warning: {
    type: 'warning',
    icon: 'exclamation triangle'
  },
  error: {
    type: 'error',
    icon: 'x icon'
  },
  change: {
    type: 'change',
    icon: 'pencil alternate'
  },
  custom: {
    type: 'custom',
    icon: 'wrench'
  }
};

class Notifications extends Component {
  constructor(props) {
    super(props);

    this.hideNotification = this.hideNotification.bind(this);
    this.renderNotifications = this.renderNotifications.bind(this);
    this.renderIcon = this.renderIcon.bind(this);
  }

  hideNotification(id) {
    const { removeItemNotification } = this.props;
    removeItemNotification(id);
  }

  renderIcon(type) {
    switch (type) {
      case notificationIcons.info.type:
        return notificationIcons.info.icon;
      case notificationIcons.success.type:
        return notificationIcons.success.icon;
      case notificationIcons.warning.type:
        return notificationIcons.warning.icon;
      case notificationIcons.error.type:
        return notificationIcons.error.icon;
      case notificationIcons.change.type:
        return notificationIcons.change.icon;
      case notificationIcons.custom.type:
        return notificationIcons.custom.icon;
      default: return null;
    }
  }


  renderNotifications() {
    const { notifications } = this.props;
    return notifications.map(({
      id,
      type,
      message,
      title
    }) => {
      const iconName = this.renderIcon(type);
      return (
        <div
          className={cx(CN, type)}
          id={id}
          key={id}
          color={type}
          onClick={() => this.hideNotification(id)}
        >
          <div className={`${CN}__ntf-box`}>
            <div>
              <i className={`${CN}__icon icon ${iconName}`} />
            </div>
            <div className={`${CN}__notification-message-box`} role="alert">
              {title && <h4 className={`${CN}__title`}>{title}</h4>}
              <div className={`${CN}__message`}>{message}</div>
            </div>
          </div>
        </div>
      );
    });
  }

  render() {
    const {
      notifications
    } = this.props;

    if (notifications.length === 0) {
      return null;
    }
    return (
      <div className={`${CN}__wrapper`}>
        {this.renderNotifications()}
      </div>
    );
  }
}

Notifications.propTypes = {
  notifications: PropTypes.array.isRequired,
  removeItemNotification: PropTypes.func.isRequired
};

export default Notifications;
