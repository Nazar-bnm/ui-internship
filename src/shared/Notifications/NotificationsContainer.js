import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Notifications from './Notifications';
import { isHideNotification } from '@/actions/actionsNotifications';

const mapStateToProps = (state) => ({
  isShowNotification: state.reducerNotifications.isShowNotification,
  isHideNotification: state.reducerNotifications.isHideNotification,
  type: state.reducerNotifications.type,
  title: state.reducerNotifications.title,
  message: state.reducerNotifications.message
});

const mapDispatchToProps = (dispatch) => ({
  isHideNotification: bindActionCreators(isHideNotification, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(Notifications);
