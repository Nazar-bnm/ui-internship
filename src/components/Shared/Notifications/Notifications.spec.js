import React from 'react';
import { shallow } from 'enzyme';

import Notifications from './Notifications';

describe('<Notifications />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      notifications: [
        {title: "warning", message: "Hey, be careful", type: "warning", id: 7986713631788149},
        {title: "warning", message: "Hey, be careful", type: "warning", id: 7986713631788148},
        {title: "warning", message: "Hey, be careful", type: "warning", id: 7986713631788147}
      ],
      removeItemNotification: jest.fn()
    }
    wrapper = shallow(<Notifications {...props} />);
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('removeItemNotification have beent called', () => {
    const inst = wrapper.instance();
    const spy = jest.spyOn(inst, 'hideNotification');
    inst.hideNotification();
    expect(spy).toBeCalled();
   })

});