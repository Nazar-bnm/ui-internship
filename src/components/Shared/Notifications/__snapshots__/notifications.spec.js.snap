// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notifications /> should match snapshot 1`] = `
<div
  className="notifications info"
  color="info"
  onClick={[Function]}
>
  <div>
    <h4
      className="notifications__title"
    >
      Info
    </h4>
    <div
      className="notifications__notification-message-box"
      role="alert"
    >
      <div
        className="notifications__message"
      >
        Information is processing
      </div>
    </div>
  </div>
</div>
`;
